spec/controllers/password_retrieval_controller_spec.rb

```diff
@@ -0,0 +1,75 @@
+require 'rails_helper'
+
+describe PasswordRetrievalController do
+  describe "password reset" do
+    it "create new entry in password_resets table" do
+      @user = User.new
+      @user.email = "example@example.edu"
+      @user.name = "ex"
+      @user.save!
+      post :send_password, {user: {email: "example@example.edu"}}
+      expect(PasswordReset.where(user_email: "example@example.edu")).to exist
+    end
+    it "modifies the token in password_resets_table" do
+      @user = User.new
+      @user.email = "example@example.edu"
+      @user.name = "Shubham"
+      @user.save!
+      @password_retrival = PasswordReset.new
+      @local_token = "some random string"
+      @password_retrival.token = @local_token
+      @password_retrival.user_email = "example@example.edu"
+      @password_retrival.save!
+      post :send_password, {user: {email: "example@example.edu"}}
+      expect(PasswordReset.find_by(user_email: "example@example.edu").token).not_to eq(@local_token)
+    end
+    it "if no user no entry is created" do
+      @user = User.new
+      @user.email = "aexample@example.edu"
+      @user.name = "Shubham"
+      @user.save!
+      post :send_password, {user: {email: "example@example.edu"}}
+      expect(PasswordReset.where(user_email: "example@example.edu")).not_to exist
+    end
+  end
+
+  describe "check if token is expired" do
+    it "checks if token is not expired" do
+      local_token = "some random string"
+      @password_retrival = PasswordReset.new
+      @password_retrival.token = local_token
+      @password_retrival.user_email = "example@example.edu"
+      @password_retrival.save!
+
+      Timecop.freeze(Date.today + 2.days) do
```

Comment:
@yangsong8 The issue with the counter may be that we used a class variable @@attemps instead of using an instance variable and adding a getter to the class.